File: sprint6/By lessons/lesson4code2/lesson4code2.cs
List<int> color; // Список цветов вершин.

void DFS(int v, List<List<int>> adjList) { // v - номер вершины
    color[v] = 1; // Вершина посещена, но ещё не обработана.
    foreach (int w in adjList[v]) { // Перебираем смежные вершины.
        if (color[w] == 0) { // Если вершина не посещена, то
            DFS(w, adjList); // запустим обход от найденной смежной вершины.
        }
    }
    color[v] = 2; // Теперь вершина обработана.
}

void mainDFS(int V, List<List<int>> adjList) {
    color = Enumerable.Repeat(0, V).ToList(); // Инициализируем список цветов.
    for (int i = 0; i < V; ++i) { // Перебираем варианты стартовых вершин.
        if (color[i] == 0) {
            DFS(i, adjList); // Запускаем обход, стартуя с i-й вершины.
        }
    }
}

File: sprint6/By lessons/lesson4code2/lesson4code2.swift
enum Color {
    case white
    case gray
    case black
}

var color: [Color] = [] // Массив цветов вершин.

func DFS(_ v: Int, _ adjList: [[Int]]) { // v - номер вершины
    color[v] = .gray // Вершина посещена, но ещё не обработана.
    for w in adjList[v] { // Перебираем смежные вершины.
        if color[w] == .white { // Если вершина не посещена, то
            DFS(w, adjList) // запустим обход от найденной смежной вершины.
        }
    }
    color[v] = .black // Теперь вершина обработана.
}

func mainDFS(_ V: Int, _ adjList: [[Int]]) {
    color = Array(repeating: .white, count: V) // Инициализируем массив цветов.
    for i in 0..<V { // Перебираем варианты стартовых вершин.
        if color[i] == .white {
            DFS(i, adjList) // Запускаем обход, стартуя с i-й вершины.
        }
    }
}
